### PR TITLE
Add Diagnosis functions

### PR DESCRIPTION
**Pull request by dmdesimone**
Add 5 diagnosis functions

### CODE REVIEW

File: configuration/pih/liquibase/sql/create_functions.sql [hunk 1]
```diff
@@ -1945,3 +1945,131 @@ limit 1;
...
+BEGIN
+
+DECLARE ret int(11);
+
+select ogr.obs_id into ret from obs ogr 
+ left outer join obs oo on oo.obs_group_id = ogr.obs_id and oo.concept_id = concept_from_mapping('PIH','7537') and oo.voided = 0
+where ogr.encounter_id = _encounter_id
+and ogr.concept_id = concept_from_mapping('PIH','7539')
+and ogr.voided = 0
+order by ISNULL(oo.value_coded) ASC ,LOCATE('7534',retrieveConceptMapping(oo.value_coded,'PIH')) DESC, ogr.obs_id ASC
```
Comment (mseaton) [inline configuration/pih/liquibase/sql/create_functions.sql:1969]: So I guess this works if there are exactly 2 choices for order - primary and secondary - but if additional options were available we'd have to tweak it, right?

File: configuration/pih/liquibase/sql/create_functions.sql [hunk 2]
```diff
@@ -1945,3 +1945,131 @@ limit 1;
...
+#
+DROP FUNCTION IF EXISTS diagnosis_noncoded;
+#
+CREATE FUNCTION diagnosis_noncoded(_encounter_id int(11), _offset int(11)) RETURNS varchar(255)
+    DETERMINISTIC
+BEGIN
+
+DECLARE ret varchar(255);
+
+select onc.value_coded into ret from obs ogr
```
Comment (mseaton) [inline configuration/pih/liquibase/sql/create_functions.sql:2013]: This looks wrong - shouldn't this be selecting onc.value_text ?

**Review by mseaton — CHANGES_REQUESTED** [review]
Overall looks good @dmdesimone .  I think I found one bug, the rest should come out in testing, but I don't see anything else obvious in the code review.  How does this perform?

File: configuration/pih/liquibase/sql/create_functions.sql (hunk 2 above)
Comment (dmdesimone) [reply configuration/pih/liquibase/sql/create_functions.sql]: thanks!  made the change

**Pull request merged**